framework/db/src/main/java/com/cloud/utils/db/GenericDaoBase.java

```diff
@@ -1935,6 +1918,23 @@ public boolean configure(final String name, final Map<String, Object> params) th
         return builder.create();
     }
 
+    private SearchCriteria<T> checkAndSetRemovedIsNull(SearchCriteria<T> sc) {
+        if (_removed != null) {
+            if (sc == null) {
+                sc = createSearchCriteria();
+            }
+            sc.addAnd(_removed.second().field.getName(), SearchCriteria.Op.NULL);
+        }
+        return sc;
+    }
+
+    public Integer getDistinctCount(SearchCriteria<T> sc, boolean removed) {
+        if (!removed) {
```

Comment:
`removed` & `_removed` seems to be redundant here & wherever used in similar way, can you directly set / unset `_removed` as per its usage ? If the purpose is to include / exclude removed records, you can use methods with naming _*IncludingRemovedBy()_ [for ex: _getDistinctCountIncludingRemovedBy()_] to be in sync with the other methods.

Reply:
As of now `getDistinctCount` returns including removed so changing it would require changing where it's been referenced, would it be better to create `getDistinctCountExcludingRemoved`  ?

Reply:
_getDistinctCountExcludingRemoved_ is not in line with other methods, so I do not recommend.

Reply:
Default method should not include removed records, explicitly mentioned methods "_*IncludingRemovedBy()_" should.

Reply:
Added! Thanks @sureshanaparti

Reply:
Perfect, changes LGTM.